### PR TITLE
[Doc] Unify the minimal versions required for PyTorch/TensorFlow/MXNet

### LINKED ISSUE #4179: [Doc] Unify the version requirements for PyTorch/TensorFlow/MXNet
About the minimal PyTorch version we are supporting:

1. Release notes and [installation page](https://www.dgl.ai/pages/start.html) write: `DGL works with PyTorch 1.9.0+, Apache MXNet 1.6+, and TensorFlow 2.3+.`
2. [docs.dgl.ai](https://docs.dgl.ai/install/index.html#backends) writes: `The required PyTorch version is 1.5.0 or later.`
3. In source code: https://github.com/dmlc/dgl/blob/150e92736fa3738504f7f4683c7f34f5f605d907/python/dgl/backend/pytorch/tensor.py#L17-L19

I think we should unify them and remove some outdated version checks, e.g.,
https://github.com/dmlc/dgl/blob/150e92736fa3738504f7f4683c7f34f5f605d907/python/dgl/dataloading/dataloader.py#L993-L1005

### PR DESCRIPTION
**Pull request by yaox12**
## Description

This resolves #4179 by unifying the minimal PyTorch/TensorFlow/MXNet versions we are supporting:
- PyTorch 1.9.0+
- MXNet 1.6+
- TensorFlow 2.3.0+

`python/dgl/_dataloading/pytorch/dataloader.py` is legacy code so I leave it untouched.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [x] Related issue is referred in this PR

### CODE REVIEW

**Review by jermainewang — APPROVED** [review]

**Comment by jermainewang** [comment]
@BarclayII  Could you also take a look at this update?

**Comment by yaox12** [comment]
Seems we need to upgrade TensorFlow to 2.3.0 in the CI docker images.

File: python/dgl/backend/tensorflow/tensor.py [hunk 1]
```diff
@@ -4,44 +4,25 @@
...
-
-    if LooseVersion(tf.__version__) < LooseVersion("2.1.0"):
-        raise RuntimeError("DGL requires tensorflow>=2.1.0.")
-
-    def zerocopy_to_dlpack(input):
-        return tfdlpack.to_dlpack(input)
-
-    def zerocopy_from_dlpack(input):
-        return tfdlpack.from_dlpack(input)
+if LooseVersion(tf.__version__) < LooseVersion("2.3.0"):
```
Comment (Rhett-Ying) [inline python/dgl/backend/tensorflow/tensor.py:15]: why do we enforce to require tf > 2.3.0 ?

**Comment by Rhett-Ying** [comment]
@jermainewang   have we decided to add below version constraints ? If so, we need to update CI image so that this PR could test and pass.

    PyTorch 1.9.0+
    MXNet 1.6+
    TensorFlow 2.3.0+

File: docker/install/conda_env/tensorflow_cpu.yml [hunk 2]
```diff
@@ -3,8 +3,7 @@ dependencies:
   - python=3.6.9
   - pip
   - pip:
-    - tensorflow==2.2.0
-    - tfdlpack
+    - tensorflow==2.3.0
```
[the remark this answers is not in the document]
Comment (yaox12) [reply docker/install/conda_env/tensorflow_cpu.yml:6]: Yes. I realized it.

File: docker/install/conda_env/tensorflow_gpu.yml [hunk 3]
```diff
@@ -4,7 +4,6 @@ dependencies:
   - pip
   - pip:
     - tensorflow==2.3.0
-    - tfdlpack-gpu
```
Comment (Rhett-Ying) [inline docker/install/conda_env/tensorflow_gpu.yml:7]: why remove this line? this package is no longer needed any more?

Comment (Rhett-Ying) [reply docker/install/conda_env/tensorflow_gpu.yml:7]: I think you'd better do not change files under //docker as the modifications you made will not take effect directly. Images used in CI remains to be the previously built ones.

**Pull request merged**